apps/web/src/components/data-table/monitor/data-table-row-actions.tsx

```diff
@@ -87,6 +89,75 @@ export function DataTableRowActions<TData>({
   }
   async function onClone() {
```

Comment:
Hi @elie222,

What are your thoughts on the recent changes in onClone? Would you prefer that I create a separate action for this? If so, could you please let me know which folder it should be placed in?

Reply:
Might got the wrong username. Hey Elie! 👋

Let's create a new `trpc` mutation function [here](https://github.com/openstatusHQ/openstatus/blob/main/packages/api/src/router/monitor.ts).

Ideally, we have your logic to clone a monitor there.

```ts
  cloneMonitorById: protectedProcedure
    .input(z.object({ id: z.number() }))
    .query(async (opts) => {
    ...
  }),
```

Edit: and use it as `api.monitor.cloneMonitorById.mutate({ id })`

Reply:
Hello @mxkaske ,
I came up with this approach. What are your thoughts on this?

Reply:
clean